apps/storybook/src/overlays/Autocomplete.stories.tsx

```diff
@@ -0,0 +1,297 @@
+    };
+
+    return (
+      <Autocomplete items={items} onInputValueChange={onInputValueChange}>
+        <AutocompleteTrigger>
+          <AutocompleteInput placeholder="Search a Language" w="208" />
+        </AutocompleteTrigger>
+        <AutocompleteContent>
+          {items.map((item, index) => (
+            <AutocompleteItem item={item} key={index}>
+              {item}
+            </AutocompleteItem>
+          ))}
+          {items.length === 0 ? (
+            <AutocompleteEmpty>No result Found</AutocompleteEmpty>
+          ) : (
+            <></>
+          )}
+        </AutocompleteContent>
+      </Autocomplete>
+    );
+  },
+};
+
+export const WithLabel: Story = {
+  render: function WithLabel() {
+    const [items, setItems] = useState(books);
+
+    const onInputValueChange = (inputValue: string) => {
+      setItems(
+        books.filter(
+          (book: Book) =>
+            !inputValue ||
+            book.title.includes(inputValue) ||
+            book.author.includes(inputValue),
+        ),
+      );
+    };
+
+    return (
+      <Autocomplete
+        isItemDisabled={isItemDisabled}
+        itemToKey={itemToKey}
+        itemToString={itemToString}
+        items={items}
+        onInputValueChange={onInputValueChange}
+      >
+        <AutocompleteTrigger>
+          <AutocompleteLabel label="Label">
+            <AutocompleteInput placeholder="Search a book" />
+          </AutocompleteLabel>
+        </AutocompleteTrigger>
+
+        <AutocompleteContent>
+          {items.map((item, index) => (
+            <AutocompleteItem item={item} key={index}>
+              {item.title}
+            </AutocompleteItem>
+          ))}
+        </AutocompleteContent>
+      </Autocomplete>
+    );
+  },
+};
+
+export const Disabled: Story = {
+  render: function Disabled() {
+    const [items, setItems] = useState(languages);
+
+    const onInputValueChange = (inputValue: string) => {
+      setItems(
+        languages.filter(
+          (language: string) =>
+            !inputValue ||
+            language.toLowerCase().includes(inputValue.toLowerCase()) ||
+            language.toLowerCase().includes(inputValue.toLowerCase()),
+        ),
+      );
+    };
+
+    return (
+      <Autocomplete
+        disabled
+        items={items}
+        onInputValueChange={onInputValueChange}
+        value="English"
+      >
+        <AutocompleteTrigger>
+          <AutocompleteInput placeholder="Search a Language" w="208" />
+        </AutocompleteTrigger>
+        <AutocompleteContent>
+          {items.map((item, index) => (
+            <AutocompleteItem item={item} key={index}>
+              {item}
+            </AutocompleteItem>
+          ))}
+        </AutocompleteContent>
+      </Autocomplete>
+    );
+  },
+};
+
+export const DisabledItems: Story = {
+  render: function DefaultSelected() {
+    const [items, setItems] = useState(books);
+    const [value, setValue] = useState("Crime and Punishment");
```

Comment:
let's have a separate story for `Value` - also the value/onValueChange types are not correct - these are supposed to be the same as the `items[number]` type
```suggestion
    const [value, setValue] = useState(books[0]);
```